### PR TITLE
customize data fields for scalars, long and others.

### PR DESCRIPTION
**Pull request by nw13slx**
## Description
- Allow user to define integer data fields with `long_fields` and put all the exception keys to `scalar fields`.
- automatically reshape per atom / per edge scalar fields.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:

- [x] My code follows the code style of this project and has been formatted using `black`.
- [ ] I have updated the documentation (if relevant).
- [ ] I have added tests that cover my changes (if relevant).
- [ ] All new and existing tests passed.
- [ ] `example.yaml` (and other relevant `configs/`) have been updated with new or changed options.
- [ ] I have updated `CHANGELOG.md`.

### CODE REVIEW

File: nequip/data/AtomicData.py [hunk 1]
```diff
@@ -24,6 +24,17 @@
 # A type representing ASE-style periodic boundary condtions, which can be partial (the tuple case)
 PBC = Union[bool, Tuple[bool, bool, bool]]
 
+_DEFAULT_SCALAR_FIELDS: Set[str] = {
+    AtomicDataDict.ATOMIC_NUMBERS_KEY,
+    AtomicDataDict.ATOM_TYPE_KEY,
+    AtomicDataDict.BATCH_KEY,
```
Comment (Linux-cpp-lisp) [inline nequip/data/AtomicData.py:30]: ?? these are all longs

File: nequip/data/_build.py [hunk 2]
```diff
@@ -72,11 +72,7 @@ def dataset_from_config(config, prefix: str = "dataset") -> AtomicDataset:
     type_mapper, _ = instantiate(TypeMapper, prefix=prefix, optional_args=config)
 
     # Register fields:
-    register_fields(
-        node_fields=config.get("node_fields", []),
-        edge_fields=config.get("edge_fields", []),
-        graph_fields=config.get("graph_fields", []),
-    )
+    instantiate(register_fields, all_args=config)
```
Comment (Linux-cpp-lisp) [inline nequip/data/_build.py:75]: lolll

but makes sense

File: nequip/train/_key.py [hunk 3]
```diff
@@ -11,7 +11,8 @@
 TRAIN = "training"
 
 ABBREV = {
-    AtomicDataDict.TOTAL_ENERGY_KEY: "e",
+    AtomicDataDict.TOTAL_ENERGY_KEY: "E",
+    AtomicDataDict.PER_ATOM_ENERGY_KEY: "e",
```
Comment (Linux-cpp-lisp) [inline nequip/train/_key.py:15]: this is important to document in changelog, will confuse people

are longer abbrevs allowed? Ei, Etot or something?

File: nequip/data/_build.py (hunk 2 above)
Comment (nw13slx) [reply nequip/data/_build.py:75]: :woman_facepalming:  well. That's what I would do when there are more than five arguments...

File: nequip/train/_key.py (hunk 3 above)
Comment (nw13slx) [reply nequip/train/_key.py:15]:
```suggestion
    AtomicDataDict.TOTAL_ENERGY_KEY: "e",
    AtomicDataDict.PER_ATOM_ENERGY_KEY: "Ei",
```

File: nequip/data/AtomicData.py (hunk 1 above)
Comment (nw13slx) [reply nequip/data/AtomicData.py:30]: but we do want most `node_fields` to have the shape of `[n, 1]`, right? Please confirm.

Comment (Linux-cpp-lisp) [reply nequip/data/AtomicData.py:30]: > but we do want most `node_fields` to have the shape of `[n, 1]`, right? Please confirm.

yes— it is more correct.

> this is the part that I'm not sure about. What's the dimension there? if I make them [n, 1], the model does not take it well.

Has nothing to do with their datatype?

But yes, I have to fix the shape logic

**Pull request merged**